### DISCUSSION TITLE
Controls always visible in canvas?

### DISCUSSION THREAD

**Discussion by AG-58** [fabricjs/fabric.js#7163] [Q&A]
Is it possible to have scaling controls always within the canvas? So that after scaling an object beyond canvas size it is still easy to scale.

The red dots would be the ideal behaviour of the scaling corners
<img width="1792" alt="fabric" src="https://user-images.githubusercontent.com/86512774/123506434-d05fda00-d681-11eb-9431-f2591152663a.png">

**Comment by asturur** [comment]
No is not possible but you probably you can do something with the custom control api, like define a control that has a dynamic position depending on object position.
You can have some scaling controls that is more on the inner side of the image, too, like near the center of the image as an alternative

You have to do the calculation yourself anyway, i do not know how do you position things at the crossing between images and canvas

**Reply by AG-58** [reply]
Thank you for the quick response
